Student-Unity-frontend/src/app/components/all-courses/all-courses.component.ts

```diff
@@ -0,0 +1,125 @@
+import { Observable, Subscription } from 'rxjs';
+import { AllCoursesService } from 'src/app/services/all-courses.service';
+import { SignInOutService } from 'src/app/services/sign-in-out.service';
+import { CommonModule } from '@angular/common';
+import { BrowserModule } from '@angular/platform-browser';
+import { AccountService } from 'src/app/services/account.service';
+import { User } from 'src/app/models/User';
+import { Router } from '@angular/router';
+import { Course } from 'src/app/models/Course';
+
+@Component({
+  selector: 'app-all-courses',
+  templateUrl: './all-courses.component.html',
+  styleUrls: ['./all-courses.component.css']
+})
+export class AllCoursesComponent implements OnInit {
+  private _courseSearch: string = '';
+  private numberOfTerms: number = 10;
+  sub!: Subscription;
+  courses: Course[] = [];
+  terms: Course[][] = [] as Course[][];
+  filteredCourses: Course[] = [];
+  showFilteredList: boolean = false;
+  private _filterByStatus: boolean = false;
+  loggedInUser = new User();
+  privilege: boolean = false;
+
+  constructor(private router: Router, private signInOutService: SignInOutService, private allCoursesService: AllCoursesService) { }
+
+  addCourse(): void {
+    this.router.navigate(['home/addCourse']);
+  }
+
+  ngOnInit(): void {
+    this.getSignedInUser();
+    this.sub = this.allCoursesService.getAllCourses().subscribe({
+      next: courses => {
+        this.courses = courses;
+        this.filterTerms();
+      },
+      error: err => console.log(err)
+    });
+  }
+
+  get filterByStatus(): boolean {
+    return this._filterByStatus;
+  }
+
+  set filterByStatus(value: boolean) {
+    this._filterByStatus = value;
+    this.filterCourses();
+  }
+
+  set courseSearch(value: string) {
+    this._courseSearch = value;
+    this.filterCourses();
+  }
+
+  get courseSearch(): string {
+    return this._courseSearch;
+  }
+
+  filterTerms(): void {
+    for (let i = 0; i < this.numberOfTerms; i++) {
+      this.terms.push( [] );
+    }
+    for (const course of this.courses) {
+      this.terms[course.term].push( course );
+    }
+    for (let i = 0; i < this.numberOfTerms; i++) {
+      console.log(this.terms[i]);
+    }
+    console.log(this.terms);
+  }
+
+
+  reverse(event: any): void {
+    let expanded = event.target.ariaExpanded;
+    let el = event.target.children[1];
+
+    if (expanded == 'true') {
+      el.classList.remove('down');
+      el.classList.add('up');
+    } else {
+      el.classList.remove('up');
+      el.classList.add('down');
+    }
+  }
+  getSignedInUser(){
+    this.signInOutService.getSignedInUser().subscribe(
+      res => {
+        console.log(res);
+        if (res.body) {
+          this.loggedInUser = res.body;
+          if (this.loggedInUser.role === 'admin') this.privilege = true;
+          console.log(this.privilege);
+        }
+    },
+      err => {
+        console.log(err);
+      }
+      );
+  }
+
+  filterCourses(): void {
```

Comment:
i'm not sure about filtering the course on the client side

Reply:
This filtering function is only useful for separating the courses in the view, it doesn't have other uses. So one call from the back-end to get all the courses and then display them in a different view.